src/wallet/wallet.cpp

```diff
@@ -2042,6 +2048,9 @@ bool CWallet::SelectCoinsMinConf(const CAmount& nTargetValue, int nConfMine, int
         if (output.nDepth < (pcoin->IsFromMe(ISMINE_ALL) ? nConfMine : nConfTheirs))
             continue;
 
+        if (pcoin->InMempool() && pcoin->GetMempoolAncestorCount() >= nMaxAncestors)
```

Comment:
@morcos and I were discussing, I think this would make more sense to limit based on both ancestor and descendant count, something like:

```
if (pcoin->InMempool() && max(pcoin->GetMempoolAncestorCount(), pcoin->GetMempoolDescendantCount()) >= nMaxChain)
    continue;
````

Otherwise, if you have an unspent output with 1 ancestor but 25 descendants, you (needlessly) could fail on one of the early calls.